Documentation/project-docs/glossary.md

```diff
@@ -87,3 +87,70 @@ Mono has implementations (though not necessarily complete) of WinForms, ASP.NET,
 ### Xamarin
 
 Xamarin is a commercial offering for building mobile applications targeting Android, iOS and Mac OS X Store. It's based on Mono, and on iOS and Android surfaces a different API profile, called the mobile profile. The subsetting was necessary to reduce the footprint, both by shipping smaller versions of the system libraries as well as making them more linker friendly. While Mono runs on Mac OS X without Xamarin, their linker is required make the app package for the Mac App Store.  Xamarin ships a full static compiler on iOS, as the platform does not support dynamic code generation.
+
+
+## Frameworks
+
+### LINQ
+
+Introduced in .NET Framework 3.5, LINQ's goal to make data processing easier. LINQ is primarily a collection of methods that extend `IEnumerable` and `IEnumerable<T>`. LINQ is intended to be used with extension methods (added in C# 3.0 and VB 9.0 at the same time as .NET Framework 3.5 was released) allowing for a function style of programing.
+
+For example to filter an array before LINQ the following would be done:
+
+```csharp
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var odds = new int[10];
+var j = -1;
+for (int i = 0; i < arr.length; i++)
+   if (arr[i] % 2 != 0)
+      odds[j++] = arr[i];
+```
+
+Now with link that can simply be done as:
+
+```csharp
+void bool IsOdd(int n)
+{
+   return n % 2 !=0;
+}
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var oods = arr.Where(IsOdd).ToArray();
+```
+
+And if we use Lambda expressions (another new feature for C# 3.0 and VB 9.0):
+
+```csharp
+var arr = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 };
+var oods = arr.Where(n => n % 2 != 0).ToArray();
```

Comment:
I was thinking about those features. Still not sure how to word it but I think I just thought of something